src/liballoc/raw_vec.rs

```diff
@@ -65,58 +65,55 @@ impl<T, A: Alloc> RawVec<T, A> {
     /// Like `with_capacity` but parameterized over the choice of
     /// allocator for the returned RawVec.
     #[inline]
-    pub fn with_capacity_in(cap: usize, a: A) -> Self {
-        RawVec::allocate_in(cap, false, a)
+    pub fn with_capacity_in(cap: usize, a: A) -> Result<Self, A::Err> {
+        RawVec::allocate_in(cap, false, a).map_err(handle_overflow_error)
     }
 
     /// Like `with_capacity_zeroed` but parameterized over the choice
     /// of allocator for the returned RawVec.
     #[inline]
-    pub fn with_capacity_zeroed_in(cap: usize, a: A) -> Self {
-        RawVec::allocate_in(cap, true, a)
+    pub fn with_capacity_zeroed_in(cap: usize, a: A) -> Result<Self, A::Err> {
+        RawVec::allocate_in(cap, true, a).map_err(handle_overflow_error)
     }
 
-    fn allocate_in(cap: usize, zeroed: bool, mut a: A) -> Self {
-        unsafe {
+    fn allocate_in(cap: usize, zeroed: bool, mut a: A) -> Result<Self, CollectionAllocErr<A::Err>> {
+       unsafe {
             let elem_size = mem::size_of::<T>();
 
-            let alloc_size = cap.checked_mul(elem_size).unwrap_or_else(|| capacity_overflow());
-            alloc_guard(alloc_size).unwrap_or_else(|_| capacity_overflow());
+            let alloc_size = cap.checked_mul(elem_size)
+                .ok_or(CollectionAllocErr::CapacityOverflow)?;
+            alloc_guard(alloc_size)?;
 
             // handles ZSTs and `cap = 0` alike
             let ptr = if alloc_size == 0 {
                 NonNull::<T>::dangling()
             } else {
                 let align = mem::align_of::<T>();
                 let layout = Layout::from_size_align(alloc_size, align).unwrap();
-                let result = if zeroed {
+                if zeroed {
                     a.alloc_zeroed(layout)
                 } else {
                     a.alloc(layout)
-                };
-                match result {
-                    Ok(ptr) => ptr.cast(),
-                    Err(_) => handle_alloc_error(layout),
-                }
+                }.map_err(CollectionAllocErr::AllocErr)?.cast()
             };
 
-            RawVec {
+            Ok(RawVec {
                 ptr: ptr.into(),
                 cap,
                 a,
-            }
+            })
         }
     }
 }
 
-impl<T> RawVec<T, Global> {
+impl<T> RawVec<T> {
     /// Creates the biggest possible RawVec (on the system heap)
     /// without allocating. If T has positive size, then this makes a
     /// RawVec with capacity 0. If T has 0 size, then it makes a
     /// RawVec with capacity `usize::MAX`. Useful for implementing
     /// delayed allocation.
     pub const fn new() -> Self {
-        Self::new_in(Global)
+        Self::new_in(AbortAdapter(Global))
```

Comment:
```suggestion
        Self::new_in(Default::default())
```
as per calls lower down?